Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindow.launchDefaultSearch:COMMENT
Method Modifier: private     internal    
Comment:/**
 * Helper method for adding launch-search to most applications. Opens the
 * search window using default settings.
 *
 * @return true if search window opened
 */

@@ -1,9 +1,15 @@
 {
+    boolean result;
     final Callback cb = getCallback();
     if (cb == null || isDestroyed()) {
-        return false;
+        result = false;
     } else {
         sendCloseSystemWindows("search");
-        return cb.onSearchRequested();
+        result = cb.onSearchRequested();
     }
+    if (!result && (getContext().getResources().getConfiguration().uiMode & Configuration.UI_MODE_TYPE_MASK) == Configuration.UI_MODE_TYPE_TELEVISION) {
+        // On TVs, if the app doesn't implement search, we want to launch assist.
+        return ((SearchManager) getContext().getSystemService(Context.SEARCH_SERVICE)).launchAssistAction(0, null, UserHandle.myUserId());
+    }
+    return result;
 }

